apps/explorer/src/ui/ExpandableList.tsx

```diff
@@ -26,30 +80,28 @@ export function ExpandableList({
     const handleShowAllClick = () =>
         setShowAll((prevShowAll: boolean) => !prevShowAll);
 
+    let showAllText = '';
+    if (showAll) {
+        showAllText = 'Show Less';
+    } else {
+        showAllText = itemsLabel
+            ? `Show All ${items.length} ${itemsLabel}`
+            : 'Show All';
+    }
+
     return (
-        <>
-            {itemsDisplayed.map((item, index) => (
-                <div key={index}>{item}</div>
-            ))}
-            {items.length > defaultItemsToShow && (
-                <div className="mt-4 flex cursor-pointer items-center gap-1 text-steel hover:text-steel-dark">
-                    <Link
-                        variant="text"
-                        onClick={handleShowAllClick}
-                        after={
-                            <ChevronDown12
-                                height={12}
-                                width={12}
-                                className={showAll ? 'rotate-180' : ''}
-                            />
-                        }
-                    >
-                        <Text variant="bodySmall/medium">
-                            {showAll ? 'Show Less' : 'Show All'}
-                        </Text>
-                    </Link>
-                </div>
-            )}
-        </>
+        <ExpandableListContext.Provider
+            value={{
+                handleShowAllClick,
+                showAll,
+                showAllText,
+                itemsDisplayed: itemsDisplayed.map((item, index) => (
+                    <div key={index}>{item}</div>
+                )),
+            }}
+        >
+            {children || <ExpandableListItems />}
+            {items.length > defaultItemsToShow && <ExpandableListControl />}
```

Comment:
IMO if you pass `children`, then this should be your responsibility to render too, but not something you _need_ to change before merging.

Reply:
I want this to be hands off, if we do not pass children then the List will render it for us, otherwise we can style the children specifically

Reply:
This makes sense, I will change that